backend/src/tasks/test/lookingGlass.test.ts

```diff
@@ -0,0 +1,671 @@
+      scanId: scan.id,
+      scanName: 'scanName',
+      scanTaskId: 'scanTaskId'
+    });
+    const vulns = await Vulnerability.find({
+      where: { domain },
+      relations: ['service']
+    });
+    expect(vulns).toHaveLength(1);
+
+    // These fields should stay the same
+    expect(vulns[0].title).toEqual('Looking Glass Data');
+    expect(vulns[0].id).toEqual(vulnerability.id);
+    expect(vulns[0].cpe).toEqual(vulnerability.cpe);
+    expect(vulns[0].state).toEqual(vulnerability.state);
+    expect(vulns[0].source).toEqual(vulnerability.source);
+    console.log(vulns[0].structuredData['lookingGlassData']);
+    // These fields should be updated
+    expect(
+      new Date(vulns[0].structuredData['lookingGlassData'][0].lastSeen)
+    ).toEqual(new Date(Today.getTime() - 7 * 24 * 60 * 60 * 1000));
+    expect(vulns[0].structuredData['lookingGlassData'][0].right_name).toEqual(
+      'Vulnerable Product - CiscoWebVPN'
+    );
+    expect(vulns[0].updatedAt).not.toEqual(vulnerability.updatedAt);
+  });
+
+  test('Merge duplicates succesfully', async () => {
+    const domain = await Domain.findOne({ id: domains[1].id });
+    const vulnerability = await Vulnerability.create({
+      domain,
+      cve: null,
+      lastSeen: new Date(Date.now()),
+      cpe: null,
+      title: `Looking Glass Data`,
+      description: '123',
+      state: 'closed',
+      structuredData: {
+        lookingGlassData: [
+          {
+            firstSeen: Today.getTime() - 25 * 24 * 60 * 60 * 1000,
+            lastSeen: Today.getTime() - 22 * 24 * 60 * 60 * 1000,
+            sources: ['Shodan Inferred Vulnerability'],
+            ref_type: 'associated-with',
+            ref_right_type: 'threat',
+            ref_right_id: 'ThreatID_1234',
+            ref_left_type: 'ipv4',
+            ref_left_id: 1234509876,
+
+            right_ticScore: 50,
+            right_classifications: ['Vulnerable Service'],
+            right_name: 'Scary Malware',
+
+            left_type: 'ipv4',
+            left_ticScore: 50,
+            left_name: '100.123.100.123',
+            vulnOrMal: 'Malware'
+          }
+        ]
+      },
+      substate: 'remediated',
+      source: 'lookingGlass'
+    }).save();
+    nock('https://delta.lookingglasscyber.com', {
+      reqheaders: {
+        Authorization: 'Bearer ' + process.env.LG_API_KEY
+      }
+    })
+      .get(
+        '/api/v1/workspaces/' +
+          encodeURIComponent(process.env.LG_WORKSPACE_NAME!) +
+          '/collections'
+      )
+      .reply(200, org_Response);
+
+    nock('https://delta.lookingglasscyber.com', {
+      reqheaders: {
+        Authorization: 'Bearer ' + process.env.LG_API_KEY
+      }
+    })
+      .post('/api/graph/query', {
+        period: 'all',
+        query: [
+          'and',
+          ['=', 'type', ['associated-with']],
+          ['or', ['=', 'right.type', 'threat'], ['=', 'left.type', 'ipv4']],
+          ['=', 'left.collectionIds', 'Collection_Id_1']
```

Comment:
Can you also add a test that ensures that the oldest firstSeen and newest lastSeen are used when we have multiple records from the lookingglass response?